doc/compute-job-configuration.md

```diff
@@ -0,0 +1,390 @@
+>
+> Due to nature of field/value storage in the default datastore (mongodb) a specific schema is used with abbreviated field names for storage optimization. For example:
+>
+>```
+>date->dt
+>availability profile->ap
+>```
+
+The following options define a set of mappings to shorter datastore fieldnames and is recommended not to be changed. Will be removed in further editions.
+
+- `e_map={fieldname1},{fieldname2}...,{fieldnameN}`
+
+  When storing endpoint a/r results in mongodb compute engine uses the above field map to store the results using abbreviated fields. the default value is `e_map=dt,ap,p,s,n,hs,a,r,up,u,d,m,pr,ss,cs,i,sc`, where dt->date,a->availability etc... (recommended not to be changed)
+
+- `s_map={fieldname1},{fieldname2}...,{fieldnameN}`
+
+  When storing service a/r results in mongodb compute engine uses the above field map to store the results using abbreviated fields. the default value is `s_map=dt,ap,p,s,n,sf,a,r,up,u,d,m,pr,ss,cs,i,sc`, where dt->date,a->availability etc... (recommended not to be changed)
+
+- `sd_map={fieldname1},{fieldname2}` (Required)
+
+  When storing status detailed results in mongodb compute engine uses the above field map to store the results using abbreviated fields. the default value is `sd_map=ts,s,sum,msg,ps,pts,di,ti`, where ts->timestamp, msg->message etc... (recommended not to be changed)
+
+- `n_eg={STRING}`
+
+  Endpoint group name type used in status detailed calculations. For e.g. `n_eg=site` if site is used as a group type
+
+- `n_gg={STRING}`
+
+  Group of groups name type used in status detailed calculations
+
+- `n_alt={STRING}`
+
+	Mapping of alternative grouping parameter used in status detail calc.
+
+- `n_altf={STRING}`
+
+  Mapping of alternative grouping parameter used in status detail calc.
+
+### /etc/ar-compute/
+
+As mentioned above secondary configuration files used by the compute-engine are stored to the /etc/ar-compute directory. Here are files describing the set of status state types used in the monitoring engine, algorithmic operations of how to combine those states, availability profiles & configuration files for available jobs.
+
+#### {TENANT_NAME}_ops.json (per tenant)
+
+These are configuration files expressed in JSON, which describe the available status types encountered in the monitoring environment of a tenant and also the available algorithmic operations to use in status aggregations.
+
+For example if the tenant name is `T1` the corresponding ops filename will be `T1_ops.json`
+
+During computations many operations take place among service statuses which need to be described explicitly. The ARGO Compute Engine gives the flexibility to the end user to declare the available monitoring statuses that are produced by the monitoring infrastructure and to map them to its internal statuses.[](Comment: Needs review by @kaggis) Then, using _truth tables_ the user can describe the logical operations on these statuses and their results. An ops file contains:
```

Comment:
As i replied above, CE at the moment needs only to be told the following:
- what kind of status type should be placed on timelines when downtimes are applied (`"default_down":"DOWNTIME"`)
- what kind of status type should be placed on timelines when recomputation exclusions are applied (`"default_unknown":"UKNOWN"`)
- what kind of status type should be placed on timelines when metric results are missing for a period of time (`"default_missing":"MISSING"`)